### ISSUE TITLE
Segment.crop and rounding errors

### ISSUE THREAD

**Issue by geniki** [pyannote/pyannote-core#25]
The `fixed` argument in the `crop` method addresses the issue of rounding/numerical precision for the duration/end of the segment. However, the same issue exists for the beginning of the segment in the `strict` mode, potentially giving your the wrong segment in `pyannote-audio`.

One way to handle this would be to add a tolerance of, say, 0.0001 when rounding. E.g.
`i = int(np.ceil(i_))`
changes to
`i = int(np.ceil(i_ + 0.0001))`

I don't think such a tolerance value would have a detrimental practical impact.

**Comment by hbredin** [comment]
Can you please provide me with a simple reproducible example where the problem occurs? 

(also: `pyannote.core.Segment` does not have a `crop` method)

**Comment by geniki** [comment]
Yes, sorry, it should have been `SlidingWindow` (and `SlidingWindowFeature` in `-audio`).

```
>>> from pyannote.core import SlidingWindow, Segment
>>> sw = SlidingWindow(duration=1.28, step=1.28, start=0)
>>> sw.crop(Segment(0, 1.28), mode="strict")
array([0]) # OK
>>> sw.crop(Segment(0 + 4 * 1.28 , 1.28 + 4 * 1.28), mode="strict")
array([4]) # OK
>>> sw.crop(Segment(0 + 469 * 1.28 , 1.28 + 469 * 1.28), mode="strict")
array([], dtype=int64) # expected 469
>>> sw.crop(Segment(0 + 469 * 1.28 , 1.28 + 469 * 1.28), mode="strict", fixed=1.28)
array([470]) # This is even worse
```
And of course, the issue is:
```
>>> (0 + 469 * 1.28) / 1.28 - 469
5.684341886080802e-14
```

My proposal above should have be `- tol_value` in `ceil` rather than `+ tol_value`.

**Comment by hbredin** [comment]
Thanks. I was able to reproduce the two errors which are indeed problematic.
We should add these as unit tests.

However, I don't understand why the proposed solution would solve the issue.
To be more precise, I think it would indeed solve this very problem but would create others (where adding the tolerance would make it jump to the previous integers)

**Comment by geniki** [comment]
Thanks for confirming the issue.

I'm probably not aware of all the use cases here so I might be missing something but my thinking is that adding `tol_value = 0.0001` (could be smaller too) and assuming a step of 1.28s (which is relatively long) would add just above 1ms offset. Since in practice we generally round to something like 0.01s steps (i.e. features are generated in 10ms steps) before we get to the `crop` step, 1ms shouldn't lead to shift to adjacent integer.

**Comment by geniki** [comment]
I just noticed that there is `SEGMENT_PRECISION = 1e-6` defined in `segment.py` so this concept seems to already be in place.
